api/v1alpha4/kubemarkmachine_types.go

```diff
@@ -47,6 +47,9 @@ type KubemarkMachineSpec struct {
 	// will be passed to the `kubemark` binary.
 	// +optional
 	KubemarkOptions KubemarkProcessOptions `json:"kubemarkOptions,omitempty"`
+
+	// KubemarkClusterSecretRef is a reference to a secret with a kubeconfig for external cluster used for kubemark pods.
+	KubemarkClusterSecretRef *corev1.ObjectReference `json:"kubemarkClusterSecretRef,omitempty"`
```

Comment:
i wonder if we should call this `KubemarkHollowPodClusterSecretRef` or something similar?

my concern is that people might be confused about what kubeconfig is supposed to go here.

Reply:
yes, I understand it might be confusing, and agree to a name change. when working with kubemark in the past, as a user, I saw general references to "kubemark cluster" in documentation meaning a cluster where kubemark hollow nodes run, but I all up for a clearer more explicit name 👍

Reply:
"cluster" is just so overloaded lol, and i hope i'm not coming off as pedantic here, i just figured since we have the opportunity to make these changes now maybe we should be explicit about things.

Reply:
👍